eng/common/TestResources/New-TestResources.ps1

```diff
@@ -611,8 +614,17 @@ try {
         }
     }
 
+    if ($UseUserCredentials) {
+        if ($TestApplicationId) {
+            Write-Warning "The specified TestApplicationId '$TestApplicationId' will be ignored when UseUserCredentials is set."
+        }
+
+        $TestApplicationOid = (Get-AzADUser -UserPrincipalName (Get-AzContext).Account).Id
```

Comment:
From some testing in the consle... If the user is NOT logged in the error message may look like: 

```
Get-AzADUser: Cannot bind argument to parameter 'UserPrincipalName' because it is an empty string.
```

Something like this gives a different error message: 

```suggestion 
        $azContext = Get-AzContext
        if (!$azContext -or $azContext.Account.Type -ne 'User') { 
            Write-Error "User must be logged in when specifying -UseUserCredentials"
        } 
        $TestApplicationOid = (Get-AzADUser -UserPrincipalName $azContext.Account).Id
```

Reply:
Wouldn't this only be an issue if there was a provisioner set as well as the flag?   Otherwise, all of the Azure resource creation is running as the user, so it would all fail.

Reply:
I just tested the logged out flow, and I was prompted to login.

Reply:
Indeed we do check Get-AzContext earlier. This is fine to ignore.